pkg/scheduler/objects/allocation.go

```diff
@@ -159,6 +159,6 @@ func (a *Allocation) IsReleased() bool {
 }
 
 // getTaskGroup returns the task group name if set.
-func (a *Allocation) getTaskGroup() string {
+func (a *Allocation) GetTaskGroup() string {
```

Comment:
Also update the comment when you export the function, however I don't think we need it at this point with the changed handler implementation proposed below